### PR TITLE
Nihdev 407 conditional directory schema validation

### PR DESCRIPTION
**Pull request by jpuerto-psc**
Adds conditional directory support

### CODE REVIEW

**Comment by jpuerto-psc** [comment]
@j-uranic - when you have a moment, can you please review the histology directory schema and make sure it is as expected?

**Comment by jpuerto-psc** [comment]
Still need to add documentation changes

**Review by sunset666 — APPROVED** [review]

File: src/ingest_validation_tools/directory_validator.py [hunk 1]
```diff
@@ -72,10 +114,24 @@ def _get_required_allowed(dir_schema: List[Dict]) -> Tuple[List[str], List[str]]
     (['this_is_required'], ['this_is_required', 'this_is_optional'])
 
     """
-    allowed = [item["pattern"] for item in dir_schema]
+    allowed = [item["pattern"] for item in dir_schema if "dependency" not in item]
```
Comment (gesinaphillips) [inline src/ingest_validation_tools/directory_validator.py:117]: Are we weeding out sub-dependencies here? If so, are those not a concern?

**Review by gesinaphillips — APPROVED** [review]

File: src/ingest_validation_tools/directory_validator.py (hunk 1 above)
Comment (jpuerto-psc) [reply src/ingest_validation_tools/directory_validator.py:117]: I don't think that we have sub-dependency situations as of yet. If we do run into that, then we will have to reconsider how we approach this.

**Pull request merged**